src/algebra/algebra/subalgebra/basic.lean

```diff
@@ -317,6 +317,10 @@ lemma map_injective {S₁ S₂ : subalgebra R A} (f : A →ₐ[R] B)
   (hf : function.injective f) (ih : S₁.map f = S₂.map f) : S₁ = S₂ :=
 ext $ set.ext_iff.1 $ set.image_injective.2 hf $ set.ext $ set_like.ext_iff.mp ih
 
+lemma map_injective' (f : A →ₐ[R] B) (hf : function.injective f) :
+  function.injective (λ S, map S f) :=
+λ S₁ S₂, map_injective f hf
```

Comment:
Note that we already have:

* [submonoid.map_injective_of_injective](https://leanprover-community.github.io/mathlib_docs/group_theory/submonoid/operations.html#submonoid.map_injective_of_injective)
* [subgroup.map_injective](https://leanprover-community.github.io/mathlib_docs/group_theory/subgroup/basic.html#subgroup.map_injective)
* [submodule.map_injective_of_injective](https://leanprover-community.github.io/mathlib_docs/linear_algebra/basic.html#submodule.map_injective_of_injective)

All of which are stated using `function.injective`; so I don't think we even need to keep the old version.

It also looks like we should be swapping the arguments to `subalgebra.map` to match these other definitions; are you happy to do that in this PR?

Reply:
Sure! Might take a few rounds of commits to find all of the errors.